temporian/implementation/numpy/data/event.py

```diff
@@ -214,48 +214,29 @@ def to_dataframe(self) -> pd.DataFrame:
 
         Returns:
             pd.DataFrame: DataFrame created from NumpyEvent.
-
         """
+        # Creating an empty dictionary to store the data
+        data = {}
 
-        feature_names = self.feature_names
-        index_names = self.sampling.index
-        columns = index_names + feature_names + ["timestamp"]
-
-        df = pd.DataFrame(data=[], columns=columns)
+        columns = self.sampling.index + self.feature_names + ["timestamp"]
+        for column_name in columns:
+            data[column_name] = []
 
-        # append every feature to the dataframe. without index
         for index, features in self.data.items():
             timestamps = self.sampling.data[index]
 
             for i, timestamp in enumerate(timestamps):
```

Comment:
Would it be more efficient to inverse the two for loops (since the internal one currently has a dictionary query)?

Something along the lines of (the following python code was not tested).

```
for index, features in self.data.items():
	timesamples = self.sampling.data[index]
	data[timestamp_key].extends(timesamples )
	for feature in features:
		data[feature.name].extends(features.data)
	for index_key in index_keys:
		data[index_key].extends( index * len(timesamples))
```

Reply:
It is not every day you can say you implemented a 250x speed-up :)